service/service.py

```diff
@@ -133,24 +133,36 @@ def create_cart_item():
 ######################################################################
 # UPDATE AN EXISTING SHOPCART ITEM
 ######################################################################
-@app.route('/shopcart/<int:item_id>', methods=['PUT'])
-def update_cart_item(item_id):
-    """
-    Update an existing item in the cart
-    """
-    app.logger.info('Request to update shopcart item with id: %s', item_id)
-    return make_response(status.HTTP_200_OK)
+@app.route('/shopcarts/<int:customer_id>/<int:product_id>', methods=['PUT'])
+def update_cart_item(customer_id, product_id):
+    app.logger.info('Request to update shopcart item with customer_id: %s, product_id: %s', customer_id, product_id)
+    cart_item = Shopcart.find_by_customer_id_and_product_id(customer_id, product_id)
+    if not cart_item:
+        app.logger.info("Customer id and product id for update have not been found")
+        return jsonify("customer id and product id for update have not been found"), status.HTTP_400_BAD_REQUEST
+
+    requested_quantity = int(request.get_json()["quantity"])
+    # bounds check
+    if requested_quantity < 0:
+        app.lgoger.info('Negative quantity requested')
```

Comment:
typo here

Reply:
logger

Reply:
also I think the minimal quantity should be 1 not 0

Reply:
yeah thx what happened was I added bunch of logger statements before pushing. I just changed the quantity too.